com.unity.render-pipelines.core/Runtime/RenderGraph/RenderGraph.cs

```diff
@@ -492,21 +732,61 @@ void PreRenderPassSetRenderTargets(in RenderPass pass, RenderGraphContext rgCont
             }
         }
 
-        void PreRenderPassExecute(int passIndex, in RenderPass pass, RenderGraphContext rgContext)
+        void PreRenderPassExecute(in CompiledPassInfo passInfo, ref RenderGraphContext rgContext)
         {
-            // TODO merge clear and setup here if possible
-            m_Resources.CreateAndClearTexturesForPass(rgContext, pass.index, pass.textureWriteList);
-            PreRenderPassSetRenderTargets(pass, rgContext);
+            // TODO RENDERGRAPH merge clear and setup here if possible
+            RenderGraphPass pass = passInfo.pass;
+
+            // TODO RENDERGRAPH remove this when we do away with auto global texture setup
+            // (can't put it in the profiling scope otherwise it might be executed on compute queue which is not possible for global sets)
             m_Resources.PreRenderPassSetGlobalTextures(rgContext, pass.textureReadList);
+
+            foreach (var texture in passInfo.textureCreateList)
+                m_Resources.CreateAndClearTexture(rgContext, texture);
+
+            PreRenderPassSetRenderTargets(passInfo, rgContext);
+
+            // Flush first the current command buffer on the render context.
+            rgContext.renderContext.ExecuteCommandBuffer(rgContext.cmd);
+            rgContext.cmd.Clear();
+
+            if (pass.enableAsyncCompute)
+            {
+                CommandBuffer asyncCmd = CommandBufferPool.Get(pass.name);
+                asyncCmd.SetExecutionFlags(CommandBufferExecutionFlags.AsyncCompute);
+                rgContext.cmd = asyncCmd;
+            }
+
+            // Synchronize with graphics or compute pipe if needed.
+            if (passInfo.syncToPassIndex != -1)
+            {
+                rgContext.cmd.WaitOnAsyncGraphicsFence(m_CompiledPassInfos[passInfo.syncToPassIndex].fence);
+            }
         }
 
-        void PostRenderPassExecute(int passIndex, in RenderPass pass, RenderGraphContext rgContext)
+        void PostRenderPassExecute(CommandBuffer mainCmd, ref CompiledPassInfo passInfo, ref RenderGraphContext rgContext)
         {
+            RenderGraphPass pass = passInfo.pass;
+
+            if (passInfo.needGraphicsFence)
+                passInfo.fence = rgContext.cmd.CreateAsyncGraphicsFence();
+
+            if (pass.enableAsyncCompute)
+            {
+                // The command buffer has been filled. We can kick the async task.
+                rgContext.renderContext.ExecuteCommandBufferAsync(rgContext.cmd, ComputeQueueType.Background);
```

Comment:
I am not sure what is happening in Unity behind the scene, but we are losing the option to chose a different queue type here vs non-rendergraph

Reply:
Will add ComputeQueueType to the API in the next PR :)